backend/src/common/pipes/validation.pipe.ts

```diff
@@ -0,0 +1,24 @@
+import { PipeTransform, Injectable, ArgumentMetadata, BadRequestException } from '@nestjs/common';
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+
+@Injectable()
+export class ValidationPipe implements PipeTransform<any> {
+  async transform(value, { metatype }: ArgumentMetadata) {
+    if (!metatype || !this.toValidate(metatype)) {
+      return value;
+    }
+    const object = plainToClass(metatype, value);
+    const errors = await validate(object);
+    console.log(errors);
+    if (errors.length > 0) {
+      throw new BadRequestException('Validation failed');
```

Comment:
čia reikėtų ištraukt tinkamesnį, ne tokį abstraktų exception, nes kiek suprantu, kviečiančioj funkcijoj gaudai visus exceptionus ir iš to darai išvadas